From 98daccf6eaf2fbf0316a06941d65b8d0417eda67 Mon Sep 17 00:00:00 2001
From: Hongzhang Yang <Hongzhang.Yang@freescale.com>
Date: Wed, 5 Sep 2012 18:09:27 +0800
Subject: [PATCH 346/449] ENGR00223123 VPU unit test: set chromaInterleave as
 default option

Set chromaInterleave as default option because it's faster than
non-chromaInterleave

Signed-off-by: Hongzhang Yang <Hongzhang.Yang@freescale.com>
---
 test/mxc_vpu_test/dec.c  |  2 +-
 test/mxc_vpu_test/main.c | 12 +++++++-----
 2 files changed, 8 insertions(+), 6 deletions(-)

diff --git a/test/mxc_vpu_test/dec.c b/test/mxc_vpu_test/dec.c
index 309104d..b2a597a 100644
--- a/test/mxc_vpu_test/dec.c
+++ b/test/mxc_vpu_test/dec.c
@@ -2083,7 +2083,7 @@ decoder_open(struct decode *dec)
 
 	ret = vpu_DecOpen(&handle, &oparam);
 	if (ret != RETCODE_SUCCESS) {
-		err_msg("vpu_DecOpen failed\n");
+		err_msg("vpu_DecOpen failed, ret:%d\n", ret);
 		return -1;
 	}
 
diff --git a/test/mxc_vpu_test/main.c b/test/mxc_vpu_test/main.c
index 92f0f24..1007cc2 100644
--- a/test/mxc_vpu_test/main.c
+++ b/test/mxc_vpu_test/main.c
@@ -76,7 +76,7 @@ char *usage = "Usage: ./mxc_vpu_test.out -D \"<decode options>\" "\
 	       "  -a <frame rate> display framerate \n "\
 	       "	default is 30 \n "\
 	       "  -t <chromaInterleave> CbCr interleaved \n "\
-	       "        default is none-interleave(0). \n "\
+	       "        default is interleave(1). \n "\
 	       "  -s <prescan/bs_mode> Enable prescan in decoding on i.mx5x - 1. enabled \n "\
 	       "        default is disabled. Bitstream mode in decoding on i.mx6  \n "\
 	       "        0. Normal mode, 1. Rollback mode \n "\
@@ -110,7 +110,7 @@ char *usage = "Usage: ./mxc_vpu_test.out -D \"<decode options>\" "\
 	       "  -g <gop size> \n "\
 	       "	default is 0 \n "\
 	       "  -t <chromaInterleave> CbCr interleaved \n "\
-	       "        default is none-interleave(0). \n "\
+	       "        default is interleave(1). \n "\
 	       "  -q <quantization parameter> \n "\
 	       "	default is 20 \n "\
 	       "\n"\
@@ -122,7 +122,7 @@ char *usage = "Usage: ./mxc_vpu_test.out -D \"<decode options>\" "\
 	       "  -h <height>capture image height \n "\
 	       "	default is 144 \n "\
 	       "  -t <chromaInterleave> CbCr interleaved \n "\
-               "        default is none-interleave(0). \n "\
+               "        default is interleave(1). \n "\
 	       "\n"\
 	       "transcode options, encoder set to h264 720p now \n "\
 	       "  -i <input file> Read input from file \n "\
@@ -166,7 +166,7 @@ char *usage = "Usage: ./mxc_vpu_test.out -D \"<decode options>\" "\
 	       "  -a <frame rate> display framerate \n "\
 	       "        default is 30 \n "\
 	       "  -t <chromaInterleave> CbCr interleaved \n "\
-	       "        default is none-interleave(0). \n "\
+	       "        default is interleave(1). \n "\
 	       "  -s <prescan/bs_mode> Enable prescan in decoding on i.mx5x - 1. enabled \n "\
 	       "        default is disabled. Bitstream mode in decoding on i.mx6  \n "\
 	       "        0. Normal mode, 1. Rollback mode \n "\
@@ -303,6 +303,8 @@ parse_args(int argc, char *argv[], int i)
 {
 	int status = 0, opt, val;
 
+	input_arg[i].cmd.chromaInterleave = 1;
+
 	do {
 		opt = getopt(argc, argv, options);
 		switch (opt)
@@ -492,7 +494,7 @@ main(int argc, char *argv[])
 
 	err = vpu_GetVersionInfo(&ver);
 	if (err) {
-		err_msg("Cannot get version info\n");
+		err_msg("Cannot get version info, err:%d\n", err);
 		vpu_UnInit();
 		return -1;
 	}
-- 
1.8.0

